bigquery_etl/cli/monitoring.py

```diff
@@ -0,0 +1,180 @@
+"""bigquery-etl CLI monitoring command."""
+
+import sys
+from pathlib import Path
+from typing import Optional
+
+import click
+from bigeye_sdk.authentication.api_authentication import ApiAuth
+from bigeye_sdk.client.datawatch_client import datawatch_client_factory
+from bigeye_sdk.controller.metric_suite_controller import MetricSuiteController
+from bigeye_sdk.model.big_config import BigConfig, TableDeployment, TableDeploymentSuite
+from bigeye_sdk.model.protobuf_message_facade import (
+    SimpleMetricDefinition,
+    SimpleMetricSchedule,
+    SimpleNamedSchedule,
+    SimplePredefinedMetric,
+    SimplePredefinedMetricName,
+)
+
+from bigquery_etl.metadata.parse_metadata import Metadata
+
+from ..cli.utils import paths_matching_name_pattern, project_id_option, sql_dir_option
+from ..util import extract_from_query_path
+
+BIGCONFIG_FILE = "bigconfig.yml"
+
+
+@click.group(
+    help="""
+        Commands for managing monitoring of datasets.
+        """
+)
+@click.pass_context
+def monitoring(ctx):
+    """Create the CLI group for the monitoring command."""
+    pass
+
+
+@monitoring.command(
+    help="""
+    Deploy monitors defined in the BigConfig files to Bigeye.
+
+    Requires BigConfig credentials to be set via BIGEYE_API_CRED_FILE env variable.
+    """
+)
+@click.argument("name")
+@project_id_option()
+@sql_dir_option
+@click.pass_context
+def deploy(ctx, name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None:
+    """Deploy monitors to Bigeye."""
+    metadata_files = paths_matching_name_pattern(
+        name, sql_dir, project_id=project_id, files=["metadata.yaml"]
+    )
+
+    for metadata_file in metadata_files:
+        project, dataset, table = extract_from_query_path(metadata_file)
+        try:
+            metadata = Metadata.from_file(metadata_file)
+            if metadata.monitoring and metadata.monitoring.enabled:
```

Comment:
Is the `metadata.monitoring` needed here? If `metadata.monitoring.enabled` is `True` the previous has to be true anyways?

Reply:
`metadata.monitoring` could be set to null or {}, then `.enabled` would raise an exception